src/applications/gqrx/mainwindow.cpp

```diff
@@ -1283,23 +1285,14 @@ void MainWindow::iqFftTimeout()
     pwr_scale = 1.0 / ((float)fftsize * (float)fftsize);
 
     /* Normalize, calculate power and shift the FFT */
+    volk_32fc_magnitude_squared_32f(d_realFftData, d_fftData + (fftsize/2), fftsize/2);
+    volk_32fc_magnitude_squared_32f(d_realFftData + (fftsize/2), d_fftData, fftsize/2);
+    volk_32f_s32f_multiply_32f(d_realFftData, d_realFftData, pwr_scale, fftsize);
+    volk_32f_log2_32f(d_realFftData, d_realFftData, fftsize);
+    volk_32f_s32f_multiply_32f(d_realFftData, d_realFftData, 10 / LOG2_10, fftsize);
+
     for (i = 0; i < fftsize; i++)
     {
-
-        /* normalize and shift */
-        if (i < fftsize/2)
-        {
-            pt = d_fftData[fftsize/2+i];
-        }
-        else
-        {
-            pt = d_fftData[i-fftsize/2];
-        }
-
-        /* calculate power in dBFS */
-        pwr = pwr_scale * (pt.imag() * pt.imag() + pt.real() * pt.real());
-        d_realFftData[i] = 10.0 * log10f(pwr + 1.0e-20);
```

Comment:
I intentionally left out the `+ 1.0e-20` part of the calculation when converting it to VOLK, because the `volk_32f_log2_32f` kernel returns -127 instead of -Inf when the input is zero, as documented here:

https://github.com/gnuradio/volk/blob/aae9fed1cd15f0f1695d5adad04c65cf8bbd3cc0/kernels/volk/volk_32f_log2_32f.h#L30-L31